storm-core/src/jvm/org/apache/storm/metric/util/DataPointPopulator.java

```diff
@@ -0,0 +1,61 @@
+package org.apache.storm.metric.util;
+
+import org.apache.storm.metric.api.IMetricsConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class DataPointPopulator implements Serializable {
+  public static final Logger LOG = LoggerFactory.getLogger(DataPointPopulator.class);
+
+  private final boolean populateMapType;
+  private final String metricNameSeparator;
+
+  public DataPointPopulator(boolean populateMapType, String metricNameSeparator) {
+    this.populateMapType = populateMapType;
+    this.metricNameSeparator = metricNameSeparator;
+  }
+
+  public Collection<IMetricsConsumer.DataPoint> populateDataPoints(Collection<IMetricsConsumer.DataPoint> dataPoints) {
+    if (!populateMapType) {
+      return dataPoints;
+    }
+
+    List<IMetricsConsumer.DataPoint> populatedDataPoints = new ArrayList<>();
+
+    for (IMetricsConsumer.DataPoint dataPoint : dataPoints) {
+      populatedDataPoints.addAll(populateDataPoint(dataPoint));
+    }
+
+    return populatedDataPoints;
+  }
+
+  public Collection<IMetricsConsumer.DataPoint> populateDataPoint(IMetricsConsumer.DataPoint dataPoint) {
+    if (!populateMapType) {
+      return Collections.singletonList(dataPoint);
+    }
+
+    List<IMetricsConsumer.DataPoint> dataPoints = new ArrayList<>();
+
+    if (dataPoint.value == null) {
+      LOG.warn("Data point with name " + dataPoint.name + " is null. Discarding." + dataPoint.name);
```

Comment:
Thanks for finding. I copied the code from Ambari metrics Storm sink which uses Log4j so missed this. I'll address it.